lib/ansible/module_utils/urls.py

```diff
@@ -82,10 +82,14 @@
 # Agreement.
 
 try:
-    import urllib2
+    from urllib2 import urlopen, Request, HTTPError, URLError, BaseHandler, HTTPPasswordMgrWithDefaultRealm, HTTPSHandler, AbstractHTTPHandler, HTTPRedirectHandler, buil_opener as url_buil_opener, install_opener as url_install_opener, HTTPBasicAuthHandler
     HAS_URLLIB2 = True
-except:
+except ImportError:
     HAS_URLLIB2 = False
```

Comment:
fixed typos, not sure about urllib2 var, it really does not work as things fail waaay before fetch_url can check for it, as it is used in class defs.